File: src/283/ Move Zeroes/Solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int k = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                if (k != 0) {
                    nums[i - k] = nums[i];
                    nums[i] = 0;
                }
            } else {
                k++;
            }
        }
    }
}

File: src/283/ Move Zeroes/Solution.kt
package `283`.` Move Zeroes`

class Solution {
    fun moveZeroes(nums: IntArray): Unit {
        var zeroes = 0
        nums.indices.forEach {
            when (nums[it]) {
                0 -> zeroes++
                else -> {
                    if (zeroes != 0) {
                        nums[it - zeroes] = nums[it]
                        nums[it] = 0
                    }
                }
            }
        }
    }
}